### PR TITLE
[WIP] Added a confusion matrix plot generator.

### PR DESCRIPTION
**Pull request by vytjan**
refering to #242 
@Ann-KathrinKiessner @dcwil @HenrikBons

### CODE REVIEW

**Comment by agramfort** [comment]
why not just using https://scikit-learn.org/stable/modules/generated/sklearn.metrics.ConfusionMatrixDisplay.html#sklearn.metrics.ConfusionMatrixDisplay ?

**Comment by robintibor** [comment]
because ours much more beautiful! ;) See discussion end of https://github.com/braindecode/braindecode/issues/242 so it was considered ! But I would try to add the class names in the example

**Comment by robintibor** [comment]
We agreed that we can keep like this (with own function that shows a bit more information than scikitlearn) depending on how strong @agramfort objects to it :D

**Comment by agramfort** [comment]
I won't strongly object I am just more lazy than you ;)

>

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 1]
```diff
@@ -343,3 +343,33 @@
...
+# get the targets
+y_true = valid_set.get_metadata().target
+y_pred = clf.predict(valid_set)
+
+# generating confusion matrix
+confusion_mat = confusion_matrix(y_true, y_pred)
+
+# add class labels
+label_dict = valid_set.datasets[0].windows.event_id.items()
+labels = list(dict(sorted(list(label_dict), key=lambda kv: kv[1])).keys())
```
Comment (robintibor) [inline examples/plot_bcic_iv_2a_moabb_cropped.py:372]:
```suggestion
# sort the labels by values (values are integer class labels)
labels = list(dict(sorted(list(label_dict), key=lambda kv: kv[1])).keys())
```

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 2]
```diff
@@ -343,3 +343,33 @@
...
+# generate confusion matrices
+# get the targets
+y_true = valid_set.get_metadata().target
+y_pred = clf.predict(valid_set)
+
+# generating confusion matrix
+confusion_mat = confusion_matrix(y_true, y_pred)
+
+# add class labels
+label_dict = valid_set.datasets[0].windows.event_id.items()
```
Comment (robintibor) [inline examples/plot_bcic_iv_2a_moabb_cropped.py:371]:
```suggestion
# label_dict is class_name : str -> i_class : int
label_dict = valid_set.datasets[0].windows.event_id.items()
```

File: braindecode/visualization/confusion_matrices.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    plt.yticks(np.arange(0, height), class_names,
+               va='center',
+               fontsize=class_names_fontsize, rotation=rotate_row_labels)
+    plt.grid(False)
+    plt.ylabel('Predictions', fontsize=15)
+    plt.xlabel('Targets', fontsize=15)
+
+    # n classes is also shape of matrix/size
+    ax.text(-1.2, n_classes + 0.2, "Specificity /\nSensitivity", ha='center', va='center',
+            fontsize=13)
```
Comment (robintibor) [inline braindecode/visualization/confusion_matrices.py:178]:
```suggestion
    ax.text(-1.2, n_classes + 0.2, "Recall", ha='center', va='center',
            fontsize=13)
```

Specificity/Sensitivity made sense for pathologies but for any classes then this value is recall (could also call sensitivity, but recall maybe better as no association with medicine / sensitivity and specificity)

Comment (robintibor) [reply braindecode/visualization/confusion_matrices.py:178]: See also https://machinelearningmastery.com/precision-recall-and-f-measure-for-imbalanced-classification/ https://en.wikipedia.org/wiki/Precision_and_recall

**Comment by robintibor** [comment]
And maybe just add same code also for trialwise example, why not?

**Comment by robintibor** [comment]
Great work!

**Pull request merged**